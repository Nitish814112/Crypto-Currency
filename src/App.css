* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  background-color:transparent;
  color:black;
 
} 

body {
  height: auto;
    width: 100vw;
  font-family: Arial, sans-serif;
 background-color: whitesmoke;
  
  
}
.header-container {
  border-bottom: 2px solid #ccc;
  padding: 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.exchange-button{
    /* position: relative; 
    margin-left: 90%;; */
    height: 45px;
    width: 30%;
    margin-top: 20px;
    color: aliceblue;
    background-color:rgb(0,0,125);
    border-radius: 5px;
   
}

.relative .absolute {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0.5rem;
    /* background-color: white; */
    /* color: white; */
    border-radius: 0.25rem;
    box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
  }
  
  .relative .absolute span {
    font-size: 0.875rem;
    font-weight: 600;
  }

  
  .portfolio-text {
    font-size: 0.8rem; /* Adjust as needed */
    font-weight: bold;
  }
  .market-cap-container {
    max-height: 83%; 
    width: 85%;/* Adjust the height as needed */
    overflow: auto;
    position: relative;
    color: #ddd;
  /* overflow-y: scroll; */
    border: 1px solid transparent;
    border-radius: 8px;
    padding: 10px;
    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.2);
    background-color:transparent; /* Ensure it has a background color */
  }
  
  .market-cap-title {
    /* font-size: 1em; */

    margin-left: -9vw;
    margin-bottom: 10px;
    /* position: sticky; */
  top: -10px;
  color: #ddd;
  background-color:transparent; /* Ensure the background color matches the container */
  z-index: 1;
  padding: 10px 0; /* Add some padding for better appearance */
  text-align: center; /* Center align the text */
  border-bottom: 1px solid #ccc; 
    /* text-align: center; */
  }
  
  .market-cap-list {
    list-style-type: none;
    padding: 0;
    margin: 0;
  }
  
  .market-cap-item {
    display: flex;
    align-items: center;
    padding: 10px 0;
    border-bottom: 1px solid #eee;
  }
  
  .market-cap-item:last-child {
    border-bottom: none;
  }
  
  .market-cap-image {
    margin-right: 10px;
    height: 20px;
    width: 20px;
  }
  
  .market-cap-details {
    flex: 1;
  }
  
  .market-cap-name {
    font-weight: bold;
  }
  
  .market-cap-value {
    display: block;
    font-size: 0.9em;
    color: #555;
  }
  
  .market-cap-change {
    margin-left: auto;
    font-size: 0.9em;
  }
/* Add styles for the modal */
.modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
}

.modal-container {
  background: white;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}

.modal {
  max-width: 300px;
  width: 100%;
}
.coin-image{
  height: 50px;
  width: 50px;
}

.price-change.green {
  color: green;
}

.price-change.red {
  color: red;
}

/* Styling for legend items */
.legend-item {
  display: flex;
  align-items: center;
}

.legend-color {
  display: inline-block;
  width: 10px;
  height: 10px;
  margin-right: 5px;
}

/* Styling for the portfolio and total text */


.total-text {
  font-size: 1rem; /* Adjust as needed */
  font-weight: bold;
}

/* Ensuring proper alignment of texts */
/* .absolute {
  position: absolute;
} */

/* .top-0 {
  top: 0;
}

.left-0 {
  left: 0;
}

.right-0 {
  right: 0;
} */

/* .p-2 {
  padding: 0.5rem;
} */

.text-sm {
  font-size: 0.5rem;
  font-weight: bold;
  margin-top: 10px;
}

.text-right {
  text-align: right;
}

.w-full {
  width: 100%;
}
  


/* for responsiveness */
/* General layout */
/* General layout */
/* General layout */
/* General layout */
/* App.css */

/* App.css */
body{
  background-color: #eee;
}
.main{
  display: flex;   
  padding-top: 20px;
  padding-left: 5%;
  gap: 2%;
}
.left{
  width: 70%;
  height:80%;
  /* border-color: antiquewhite; */
  display: flex;
  flex-direction: column;
  
}
.right{ 
  width:30%;  
  height:10%;
  order: 2;
  /* background-color: aqua; */
 
}
.nav{
  width: 100%;
  height: 10%;
  
  /* background: white; */
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  /* z-index: 1000; */
  /* background-color: blueviolet; */
  /* border-bottom: 1px solid silver; */
 
}

.footer{
  width: 100%;
  height: 3%;
  display: flex;
  justify-content: center;
  border-top:1px solid transparent;
  border-radius: 8px;
  
  order: 3;
  margin-top: 1%;
  padding-top: 0.8%;
  color:rbg(100,80,80);
  font-weight: bold;
  /* background: white; */
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}
.search{
width: 100%;
height: 5%;
border-radius: 8px;

box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
display: flex;
gap: 2%;
padding: 0.8%;
background-color:transparent;


}
.search>.shadow{
  width: 10%;
  height: 2rem;
  border: 1px solid #ddd;
  background-color:transparent;

  
}
.selecting{
  position: absolute;
  /* width: fit-content; */
  margin-left: 400px;
  background: whitesmoke;
  height: 10px;
  font-size: 0.8rem;
  font-weight: bold;
  border-radius: 2px;
  border: none;
  text-align: center;
  /* background-color: whitesmoke; */
 
  
  
}

.hovering{
  
  width: fit-content;
  margin-left: 450px;
  z-index: 999;
  /* background-color: white; */
  /* margin-top: 10px */
}
.hovering:hover{
  border: 1px solid #ddd;
}
.shadow_input{
  /* color: aqua; */
  width: 90%;
  height: 2rem;
  border-radius: 8px;
  border: 1px solid transparent;
  color: #555;
  /* background-color: red; */
}

.graph{
  width: 100%;
  height: 60%;
  padding: 1%;
  background: transparent;
  color: black;
  border: 1px solid transparent;
  box-shadow: 0 4px 20px rgba(0, 0, 0, 0.2);
  border-radius: 8px; 
  /* background-color: cadetblue; */
}

.exchange_portfolio{
  width: 100%;
  height: 40%;
  display: flex;
  gap: 1%;
  padding-top: 2%;
  justify-content: center;
  align-items: center;
  background: transparent;

  /* background-color: brown; */
  }
  .currency-btn{
    border: 1px solid transparent;
    
    } 

    .currency-btn:hover{
      border: 1px solid whitesmoke;
    }

  .portfolio{
      width:50%;
      height:100%;
      border: 1px solid transparent;
      box-shadow: 0 4px 20px rgba(0, 0, 0, 0.2);
  border-radius: 8px;
  background: transparent;
      
  }
  .changeText:active{
    border: 1px solid black;
  }

  .exchange{

      width: 50%;
      height: 270px;
      padding-left: 4%;
      background: white;
      box-shadow: 0 4px 20px rgba(0, 0, 0, 0.2);
      border-radius: 8px; 
      border: 1px solid transparent;
      background: transparent;
      color: #ddd;
      
  }

  @media (max-width:1180px){
    
    .main{
      width: 1180px;
    }
    .nav{
      width: 1180px;
    }
    .footer{
      width:1180px;
    }
  }

  @media (max-width:1370px){
    .select-width{
      margin-left: 250px;  
        
    }
  }

    @media (max-width:641px){
    
      .main{
        width: 641px;
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
      }
      .graph{
       width: 150%;
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;

      }
      
      /* .selecting{
        margin-left: 325px;
      } */
      .search{
        width: 641px;
      }
      .portfolio{
        width: 641px;
      }
      .exchange{
        width: 641px ;
        margin-top: 10px;
      }
      .right{
        width: 641px;
      }
      .market-cap-container{
        width: 641px;
        margin-top: 10px;
      }
      /* .currency-btn{
      border: 1px solid transparent;
      
      } */
  
      .shadow{
        position: relative;
        width: 50px;
      }
      .shadow_input{
        margin-left: 30px;
        width: 100%;
      }

      .exchange_portfolio{
        width: 641px;
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
      }
      .nav{
        width:641px;
        border: none;
        box-shadow: none;
      }
      .footer{
        width:641px;
        border: none;
        box-shadow: none;
        margin-top: 20px;
      }
    
   
  }

  /* Center the spin div */
.spin {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh; /* Full viewport height */
  text-align: center;
  font-size: 1.5rem;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  z-index: 9999;
  background-color: rgba(255, 255, 255, 0.8); /* Optional: adds a semi-transparent background */
}

/* Style the spinner */
#spinner {
  border: 10px solid rgba(0, 0, 0, 0.1);
  border-top: 10px solid #3498db; /* Blue color for the spinner */
  border-radius: 50%;
  width: 50px;
  height: 50px;
  animation: spin 1s linear infinite;
  margin-left: 10px;
}

/* Keyframes for spinner animation */
@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.error-message {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  margin-top: -6%;
  text-align: center;
}

.error-message h2 {
  font-size: 1.5rem;
  color: #ff6347; /* Tomato color for emphasis */
  margin-bottom: 1rem;
}

.error-message button {
  background-color: #ff6347; /* Tomato color for button */
  color: white;
  border: none;
  border-radius: 5px;
  padding: 0.5rem 1rem;
  font-size: 1rem;
  cursor: pointer;
  transition: background-color 0.3s;
}

.error-message button:hover {
  background-color: #ff4500; /* Darker shade of tomato */
}

