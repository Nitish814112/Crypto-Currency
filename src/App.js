import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { cryptoDataAction } from "./component/redux/action/cryptoDataAction";
import "./App.css";
import Coin from "./component/Coin";
import CryptoChart from "./component/CryptoChart";
import Exchange_buttons from "./component/Exchange_Buttons";
import DisplayMarketCap from "./component/DisplayMarketCap";
import currencySymbols from "./component/utility/currencySymbols";
import Nav from "./component/Nav";
import { monthWiseActions } from "./component/redux/action/monthWiseAction";

function App() {
  const dispatch = useDispatch();
  const GET_CRYPTO = useSelector((state) => state.GET_CRYPTO);
  const GET_MONTH_WISE = useSelector((state) => state.GET_MONTH_WISE);

  const { loading: loadingCrypto, coins, error: errorCrypto } = GET_CRYPTO;
  const {
    loading: loadingMonthWise,
    data,
    error: errorMonthWise,
  } = GET_MONTH_WISE;
  // console.log('errorMonthWise :',errorMonthWise);

  const [overallLoading, setOverallLoading] = useState(true);
  const [overallError, setOverallError] = useState(null);
  const [selectedCurrency, setSelectedCurrency] = useState("usd");
  const [searchTerm, setSearchTerm] = useState("");
  const [searchResult, setSearchResult] = useState(null);

  useEffect(() => {
    dispatch(cryptoDataAction());
    dispatch(monthWiseActions());
  }, [dispatch]);

  useEffect(() => {
    if (!loadingCrypto && !loadingMonthWise) {
      setOverallLoading(false);
      if (errorCrypto || errorMonthWise) {
        setOverallError(true);
      } else {
        setOverallError(false);
      }
    } else {
      setOverallLoading(true);
    }
  }, [loadingCrypto, loadingMonthWise, errorCrypto, errorMonthWise]);

  const handleRetry = () => {
    setOverallError(null);
    dispatch(cryptoDataAction());
    dispatch(monthWiseActions());
  };

  const handleSearch = (e) => {
    e.preventDefault();
    const result = coins.find(
      (coin) => coin.name.toLowerCase() === searchTerm.toLowerCase()
    );
    setSearchResult(result ? result : "No coins found");
  };

  const variousCountriesCurrency = ["usd", "inr", "euro", "aed", "yuan", "ruble", "yen"];

  const closeModal = () => {
    setSearchResult(null);
  };

  return (
    <>
      {overallLoading ? (
        <h2 className="spin">
        Loading Your Dashboard... <span id="spinner"></span>
        </h2>
      ) : overallError ? (
        <div className="error-message">
          <h2>Something went wrong</h2>
          <button onClick={handleRetry}>‚ü≥</button>
        </div>
      ) : (
        <>
          
            <div className="nav">
              <Nav />
            </div>
          

          <div className="main">
            <div className="left">
              <div className="search">
                <select
                  className="shadow portfolio-text"
                  style={{fontSize:"0.8rem",paddingLeft:"15px", borderRadius: '8px'}}
                  value={selectedCurrency}
                  onChange={(e) => setSelectedCurrency(e.target.value)}
                >
                  {variousCountriesCurrency.map((currency) => (
                    <option style={{fontSize:"0.8"}} key={currency} value={currency}>
                      {currency[0].toUpperCase()+currency.slice(1)}
                    </option>
                  ))}
                </select>
                <form onSubmit={handleSearch}>
                  <input
                    className="shadow_input"
                    style={{width:"500px",padding:"15px"}}
                    type="search"
                    placeholder="üîçSearch by coin"
                    value={searchTerm}
                    onChange={(e) => setSearchTerm(e.target.value)}
                  />
                </form>
              </div>
              <div className="graph">
                <CryptoChart data={data} />
              </div>
              <div className="exchange_portfolio">
                <div className="portfolio">
                  <Coin coin={coins} />
                </div>
                <div className="exchange">
                  <Exchange_buttons coin={coins} />
                </div>
              </div>
            </div>
            <div className="right">
              <DisplayMarketCap coins={coins} selectedCurrency={selectedCurrency} />
            </div>
            
          </div>
          <div className="footer"><h1>Made with ‚ù§Ô∏è by Nitish ¬©Ô∏è  2024</h1></div>
          {searchResult && (
            <div
              className="modal-overlay fixed inset-0 flex items-center justify-center bg-black bg-opacity-50 z-50"
              onClick={closeModal}
            >
              <div
                className="modal-container relative bg-white p-6 rounded-lg shadow-lg w-11/12 max-w-lg"
                onClick={(e) => e.stopPropagation()}
              >
                <button
                  className="absolute top-2 right-2 text-gray-500 hover:text-gray-700"
                  onClick={closeModal}
                >
                  &times;
                </button>
                <div className="modal">
                  {typeof searchResult === "string" ? (
                    <p>{searchResult}</p>
                  ) : (
                    <div>
                      <img
                        src={searchResult.image}
                        alt={searchResult.name}
                        className="coin-image"
                      />
                      <h3 className="portfolio-text">{searchResult.name}</h3>
                      <p className="portfolio-text">
                      
                        Current Price: {currencySymbols[selectedCurrency]}
                        {searchResult.current_price.toLocaleString()}
                      </p>
                      <p className="portfolio-text">
                        Market Cap: {currencySymbols[selectedCurrency]}
                        {searchResult.market_cap.toLocaleString()}
                      </p>
                      <p
                        className={`price-change ${
                          searchResult.price_change_percentage_24h >= 0
                            ? "green"
                            : "red"
                        } portfolio-text`}
                      >
                        {searchResult.price_change_percentage_24h >= 0
                          ? "‚ñ≤"
                          : "‚ñº"}
                        {searchResult.price_change_percentage_24h.toFixed(2)}%
                      </p>
                    </div>
                  )}
                </div>
              </div>
            </div>
          )}
        </>
      )}
    </>
  );
}

export default App;
