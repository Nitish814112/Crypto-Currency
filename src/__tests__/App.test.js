import React from 'react';
import { Provider } from 'react-redux';
import { render, act } from '@testing-library/react';  // Import act from react
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../App';

jest.mock('../component/CryptoChart', () => () => <div>Mocked CryptoChart</div>);  // Mocking the chart component

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('App component snapshot', () => {
  let store;
  let initialState;

  beforeEach(() => {
    initialState = {
      GET_CRYPTO: { loading: false, coins: [], error: null },
      GET_MONTH_WISE: { loading: false, data: [], error: null },
    };
    store = mockStore(initialState);
  });

  test('renders correctly and matches snapshot', async () => {
    let asFragment;

    await act(async () => {
      // Wrap the render call in act
      const result = render(
        <Provider store={store}>
          <App />
        </Provider>
      );
      asFragment = result.asFragment;
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
