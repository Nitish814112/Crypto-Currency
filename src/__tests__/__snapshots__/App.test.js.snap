// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <img
      alt=""
      src="unique_logo.png"
      style="height: 90px; width: 90px; margin-left: 4%;"
    />
  </div>
  <div
    class="main"
  >
    <div
      class="left"
    >
      <div
        class="search"
      >
        <select
          class="portfolio-text changeText"
          style="font-size: 0.8rem; padding-left: 15px; border-radius: 8px;"
        >
          <option
            class="portfolio-text"
            value="usd"
          >
            Usd
          </option>
          <option
            class="portfolio-text"
            value="inr"
          >
            Inr
          </option>
          <option
            class="portfolio-text"
            value="euro"
          >
            Euro
          </option>
          <option
            class="portfolio-text"
            value="aed"
          >
            Aed
          </option>
          <option
            class="portfolio-text"
            value="yuan"
          >
            Yuan
          </option>
          <option
            class="portfolio-text"
            value="ruble"
          >
            Ruble
          </option>
          <option
            class="portfolio-text"
            value="yen"
          >
            Yen
          </option>
        </select>
        <form>
          <input
            class="shadow_input"
            placeholder="🔍Search by coin"
            style="width: 500px; padding: 15px;"
            type="search"
            value=""
          />
        </form>
      </div>
      <div
        class="graph"
      >
        <div
          class="relative inline-block text-left selecting"
        >
          <div
            class="border border-gray-300 rounded-lg p-2 cursor-pointer setText "
            style="height: 40px;"
          >
            Currency
          </div>
        </div>
        <div
          class="w-full"
        >
          <div
            class="flex items-center justify-between gap-4 mb-4 flex-nowrap"
          >
            <div
              class="flex gap-2 flex-nowrap"
            >
              <button
                class="py-1 px-3 border rounded-md bg-transparent text-gray-700 portfolio-text currency-btn"
              >
                1D
              </button>
              <button
                class="py-1 px-3 border rounded-md bg-transparent text-gray-700 portfolio-text currency-btn"
              >
                1W
              </button>
              <button
                class="py-1 px-3 border rounded-md bg-transparent text-gray-700 portfolio-text currency-btn"
              >
                1M
              </button>
              <button
                class="py-1 px-3 border rounded-md bg-transparent text-gray-700 portfolio-text currency-btn"
              >
                3M
              </button>
              <button
                class="py-1 px-3 border rounded-md bg-transparent text-yellow-700 portfolio-text currency-btn"
              >
                6M
              </button>
              <button
                class="py-1 px-3 border rounded-md bg-transparent text-gray-700 portfolio-text currency-btn"
              >
                1Y
              </button>
            </div>
            <div
              class="flex flex-col items-center w-full"
              style="color: black;"
            >
              <div
                class="flex gap-2 flex-nowrap select-curr justify-center"
              >
                <div
                  class="select-normal portfolio-text select-width css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class=" css-19q7pao-control"
                  >
                    <div
                      class=" css-hlgwow"
                    >
                      <div
                        class=" css-1dimb5e-singleValue"
                      >
                        Line
                      </div>
                      <div
                        class=" css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=""
                          id="react-select-2-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-1wy0on6"
                    >
                      <span
                        class=" css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="chart-container h-48 overflow-hidden"
            style="color: black;"
          >
            <canvas
              height="150"
              role="img"
              width="300"
            />
          </div>
        </div>
      </div>
      <div
        class="exchange_portfolio"
      >
        <div
          class="portfolio"
        >
          <div
            class="flex flex-col items-center w-full"
          />
        </div>
        <div
          class="exchange"
        >
          <div>
            <h2
              class="portfolio-text absolute"
              style="margin-top: -20px;"
            >
              Exchange Coins
            </h2>
            <div
              class="flex flex-col gap-3 ml-4 mt-8"
            >
              <div
                class="flex flex-col"
              >
                <span
                  style="margin-left: 45%;"
                >
                  Enter Value
                </span>
                <div
                  class="flex gap-3 mt-2"
                >
                  <label
                    class="portfolio-text"
                    for="sellCurrency"
                    style="color: red;"
                  >
                    Sell:
                  </label>
                  <select
                    class="portfolio-text"
                    id="sellCurrency"
                    style="border: 0.5px solid grey; width: 22%;"
                  />
                  <input
                    class="portfolio-text"
                    min="0.01"
                    style="border: 0.5px solid grey; color: red; width: 40%;"
                    type="number"
                    value="1"
                  />
                </div>
              </div>
              <div
                class="flex gap-3 mt-4"
              >
                <label
                  class="portfolio-text"
                  for="buyCurrency"
                  style="color: green;"
                >
                  Buy:
                </label>
                <select
                  class="portfolio-text"
                  id="buyCurrency"
                  style="border: 0.5px solid grey; width: 22%;"
                />
                <input
                  class="portfolio-text"
                  readonly=""
                  style="border: 0.5px solid grey; color: green; width: 40%;"
                  type="number"
                  value="0"
                />
              </div>
              <button
                class="exchange-button"
                style="margin-left: 30%; width: 140px;"
              >
                Exchange
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="right"
    >
      <div
        class="market-cap-container"
        style="height: 670px;"
      >
        <h3
          class="market-cap-title portfolio-text"
          style="color: black;"
        >
          Cryptocurrency by 
          <p>
            market cap
          </p>
        </h3>
        <ul
          class="market-cap-list"
        />
      </div>
    </div>
  </div>
  <div
    class="footer"
  >
    <h1>
      Made with ❤️ by Nitish ©️ 2024
    </h1>
  </div>
</DocumentFragment>
`;
